src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -160,33 +206,50 @@ public String getHelpFile() {
         public ParameterDefinition newInstance(StaplerRequest req, JSONObject formData) throws FormException {
             return req.bindJSON(PromotedBuildParameterDefinition.class, formData);
         }
-
+        
         /**
          * Checks the job name.
          */
         public FormValidation doCheckJobName(@AncestorInPath Item project, @QueryParameter String value ) {
+            final Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins == null) {
+                return FormValidation.error("Jenkins instance is not ready");
+            }
             if (!project.hasPermission(Item.CONFIGURE) && project.hasPermission(Item.EXTENDED_READ)) {
                 return FormValidation.ok();
             }
+            
             project.checkPermission(Item.CONFIGURE);
 
             if (StringUtils.isNotBlank(value)) {
-                AbstractProject p = Jenkins.getInstance().getItem(value,project,AbstractProject.class);
-                if(p==null)
+                // JENKINS-25011: also look for jobs in folders.
+                final AbstractProject p = ItemPathResolver.getByPath(value, project, AbstractProject.class);
+                if (p==null) {
```

Comment:
@recena spaces after brackets is apache style that is not used in most jenkins code. So oleg's variant is standard sun/oracle coding style. Changing bad coding style into the most used and better readable is good.